src/plugins/clips-executive/clips/coordination-mutex.clp

```diff
@@ -440,6 +418,19 @@
 			(assert (mutex (name ?id) (state (if ?locked then LOCKED else OPEN))
 			               (locked-by ?locked-by) (lock-time ?lock-time)))
 		)
+		(do-for-fact ((?m mutex)) (eq ?m:name ?id)
+      (if (or (eq ?m:state OPEN) (eq ?m:locked-by (cx-identity)))
+      then
+        (if (and (eq ?m:request UNLOCK) (not ?locked))
+        then
+          (modify ?m (response UNLOCKED))
+        )
+        (if (and (or (eq ?m:request LOCK) (eq ?m:request RENEW-LOCK)) ?locked)
```

Comment:
You need to check `locked-by`, otherwise another agent may get the lock and we'd still assume we acquired it!

Reply:
This is checked in row 422

Reply:
You're right, missed it.